deploy-service/common/src/main/java/com/pinterest/deployservice/handler/GoalAnalyst.java

```diff
@@ -532,6 +546,26 @@ void process(String envId, EnvironBean env, PingReportBean report, AgentBean age
             }
         }
 
+        // update host_tags table
+        if (ec2Tags != null && env != null && env.getDeploy_constraint_id() != null) {
```

Comment:
we might want to check if we need to update hostTag, if there is no difference we can just skip updating to avoid db writes.

Reply:
A very good point

Reply:
updated